Add new pages to microsite

miguelangel-dev opened the pull request:
### Details
This PR adds new pages to `First Steps` sections about:
- How to convert an Xcode Playground to nef Playground.
- How to create a Playground Book (iPad) with 3rd-party libraries.

miguelangel-dev committed:
update color selector in nav api-docs

miguelangel-dev committed:
update carbon information

miguelangel-dev committed:
microsite structure with new pages

miguelangel-dev committed:
add page to microsite: how to convert an Xcode Playground to nef Playground

miguelangel-dev committed:
add steps to tutorials

miguelangel-dev committed:
fix dash url

minor changes

miguelangel-dev committed:
add page how-to create playground book

truizlop commented on Documentation.app/Contents/MacOS/Command-Line.playground/contents.xcplayground:
```diff
@@ -1,2 +1,11 @@
+        <page name='Creating a Playground Book for iPad'/>
+        <page name='Compiling a nef Playground'/>
+        <page name='Generating a Markdown document'/>
+        <page name='Jekyll integration'/>
```
Perhaps the title of this page should be *Creating a Jekyll site* or something similar, to keep consistency in the section.

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/Pages/How to export my snippets?.xcplaygroundpage/Contents.swift:
```diff
@@ -19,11 +19,10 @@ In this tutorial, we assume you do not have any project, and we will follow up f
- &nbsp;
- ### 2. Add content to nef Playground
+ 
+ Go back to [Creating a nef Playground](/docs/command-line/creating-a-nef-playground/) to read more about these and other parameters you can use :)
```
```suggestion
 Go back to [Creating a nef Playground](/docs/command-line/creating-a-nef-playground/) to read more about these and other parameters you can use.
```

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/Pages/How to use Carbon API?.xcplaygroundpage/Contents.swift:
```diff
@@ -11,14 +11,17 @@
- Now, you can create a Carbon style to apply to our code, for the generated snippet.
+ #### Step 2: Create a Carbon style
+ 
+ Create a carbon style to apply to our code, for the generated snippet.
```
```suggestion
 Create a carbon style to apply to our code, in the generated snippet image.
```

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/Pages/Playground Book with 3rd-party libraries.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,79 @@
+/*:
+ ## 🔨 Playground Book with 3rd-party libraries
+ 
+ As you read in `Getting started` section, you can use nef in different forms. Accordingly, we will see how to create a Playground for iPad compatible with 3rd-party libraries, using the Command-Line Tool and the Xcode plugin.
```
```suggestion
 This page describes how to create a Playground for iPad with support for 3rd-party libraries, using the Command-Line Tool and the Xcode plugin.
```

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/Pages/Playground Book with 3rd-party libraries.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,79 @@
+ 
+ #### Create a Swift Package
+ 
+ Before you need to define a Swift Package with your dependencies. In this tutorial, we will create one with Bow as a dependency. It'll be located at `~/Desktop`.
```
```suggestion
 First of all, you need to define a Swift Package with your dependencies. In this tutorial, we will create one with Bow as a dependency. It'll be located at `~/Desktop`.
```

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/Pages/Playground Book with 3rd-party libraries.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,79 @@
+ 
+ ### 1. How to create a Playground book using nef Command-Line?
+ 
+ #### Step 1: Run nef with subcommand `iPad`
```
```suggestion
 #### Step 1: Run nef with subcommand `ipad`
```

miguelangel-dev replied:
not sure, is it the suggestion do you want to do?

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/Pages/Playground Book with 3rd-party libraries.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,79 @@
+ 
+ #### Step 2: Copy `Tutorial.playgroundbook` to your iPad
+ 
+ You can copy the Playground Book using `AirDrop`, `iCloud Drive` or `iTunes`. Our recommendation is use `AirDrop`; it is an easy way to transfer documents to your iPad and it will be auto-opened into Playgrounds app.
```
```suggestion
 You can copy the Playground Book using AirDrop, iCloud Drive or iTunes. Our recommendation is use AirDrop; it is an easy way to transfer documents to your iPad and it will be opened automatically into the Swift Playgrounds app.
```

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/Pages/Playground Book with 3rd-party libraries.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,79 @@
+ 
+ #### Step 3: Open it in your iPad using `Playgrounds` app
+ 
+ Once you copy `Tutorial.playgroundbook` to your iPad, you only need to find it and open. It'll be auto-opened using `Playgrounds` app.
```
```suggestion
 Once you copy `Tutorial.playgroundbook` to your iPad, you only need to find it and open. It'll be opened automatically using Swift Playgrounds app.
```

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/Pages/Playground Book with 3rd-party libraries.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,79 @@
+ 
+ #### Step 1: Install `nef Plugin` from GitHub
+ 
+ You can find the last `.dmg` release in [GitHub](https://github.com/bow-swift/nef-plugin/releases) page.
```
```suggestion
 You can find the last `.dmg` release in [GitHub](https://github.com/bow-swift/nef-plugin/releases).
```

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/Pages/Playground Book with 3rd-party libraries.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,79 @@
+ 
+ You can find the last `.dmg` release in [GitHub](https://github.com/bow-swift/nef-plugin/releases) page.
+
+ Just download the installer and move `nef` to `Applications` folder; after, you need to open it and follow the instructions:
```
```suggestion
 Just download the installer and move `nef` to the *Applications* folder; afterwards, you need to open it and follow the instructions:
```

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/Pages/Playground Book with 3rd-party libraries.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,79 @@
+ 
+ ![](/assets/package-playground-book.png)
+ 
+ After the Playground has been created, you can follow the steps from the second one that you seen in the above section.
```
```suggestion
 After the Playground has been created, you can continue from the second step in the previous section in order to load it into the iPad.
```

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,73 @@
+/*:
+ ## Xcode Playground ⇢ nef Playground
+ 
+ I assume you have an Xcode Playground and you want to take advantage of `nef Playground` features:
```
```suggestion
 Let's assume you already have an Xcode Playground and you want to take advantage of `nef Playground` features:
```

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,73 @@
+ 
+ - 🎉 Adds support for __third party libraries__.
+ 
+ - 🖥 Enables to __compile__ using command-line (out-of-the-box Apple IDE).
```
```suggestion
 - 🖥 Enables __compilation__ of Playgrounds using command-line.
```

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,73 @@
+ 
+ - 🖥 Enables to __compile__ using command-line (out-of-the-box Apple IDE).
+ 
+ + 🗞 Renders into __markdown__ files, __Jekyll__ site or __Carbon__ snippets.
```
```suggestion
 + 🗞 Renders into __markdown__ files, a __Jekyll__ site or __Carbon__ snippets.
```

miguelangel-dev committed:
fix dash button

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,73 @@
+ 
+ #### Step 1: Prepare your Xcode Playground
+ 
+ Given an Xcode Playground, which name will be `MyPlayground` and it is located at `~/Desktop`
```
```suggestion
 Let's assume you have an Xcode Playground, named `MyPlayground` in this example, and located at `~/Desktop`.
```

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,73 @@
+ 
+ Given an Xcode Playground, which name will be `MyPlayground` and it is located at `~/Desktop`
+ 
+ If your playground does not have any page, add someone, for example `Tutorial`. Your playground structure should look like:
```
```suggestion
 If your playground does not have any page, add some; for example, a page titled `Tutorial`. Your playground structure should look like:
```

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift:
````diff
@@ -0,0 +1,73 @@
+ ➜ nef playground --playground ~/Desktop/MyPlayground.playground --name MyPlayground
+ ```
+ 
+ It'll create a `nef Playground` in your current location which name will be `MyPlayground`. You can go to the section **Command-Line** to learn more about nef Playgrounds.
````
```suggestion
 It'll create a `nef Playground` in your current location, named `MyPlayground`. You can go to the section **Command-Line** to learn more about nef Playgrounds.
```

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,73 @@
+ 
+ It'll create a `nef Playground` in your current location which name will be `MyPlayground`. You can go to the section **Command-Line** to learn more about nef Playgrounds.
+ 
+ Using one of these options, you can change the dependency manager and its dependencies by default
```
```suggestion
 Using one of these options, you can change the dependency manager and its default dependencies:
```

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,73 @@
+ 
+ Using one of these options, you can change the dependency manager and its dependencies by default
+
+ - `--podfile <podfile>` Specify a **Podfile** with your own dependencies. Create a `Podfile` listing your dependencies.
```
```suggestion
 - `--podfile <podfile>`: Supply a **Podfile** with your own dependencies. You need to create a `Podfile` listing your dependencies.
```

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,73 @@
+
+ - `--podfile <podfile>` Specify a **Podfile** with your own dependencies. Create a `Podfile` listing your dependencies.
+
+ - `--cartfile <cartfile>` Specify a **Cartfile** with your dependencies. Create a Cartfile listing your dependencies.
```
```suggestion
 - `--cartfile <cartfile>`: Supply a **Cartfile** with your dependencies. You need to create a Cartfile listing your dependencies.
```

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,73 @@
+ 
+ ### How to make an Xcode Playground compatible with CocoaPods?
+ 
+ Following the options seen above, we will continue the example to make our Xcode Playground compatibles with 3rd-party libraries using CocoaPods as a dependency manager.
```
```suggestion
 Following the options seen above, we will continue the example to make our Xcode Playgrounds compatible with 3rd-party libraries using CocoaPods as a dependency manager.
```

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,73 @@
+ 
+ #### Step 1: Create a `Podfile` listing your dependencies
+
+ I'll be named `MyPodfile` and located at `~/Desktop`
```
```suggestion
 It will be named `MyPodfile` and located at `~/Desktop`
```

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift:
````diff
@@ -0,0 +1,73 @@
+ end
+ ```
+ 
+ #### Step 2: Converts Xcode Playground using your dependencies
````
```suggestion
 #### Step 2: Convert an Xcode Playground using your dependencies
```

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift:
````diff
@@ -0,0 +1,73 @@
+ ➜ nef playground --playground ~/Desktop/MyPlayground.playground --name MyPlayground --podfile ~/Desktop/MyPodfile
+ ```
+ 
+ > Currently, nef is compatible with CocoaPods and Carthage. In the future, when Apple fixes a known issues in Swift package and Xcode ([#47668990](https://github.com/bow-swift/nef/issues/33)) it'll also be compatible with SPM.
````
```suggestion
 > Currently, nef is compatible with CocoaPods and Carthage. In the future, when Apple fixes a known issues in Swift Package Manager and Xcode ([#47668990](https://github.com/bow-swift/nef/issues/33)), it will also be compatible with SPM.
```

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/contents.xcplayground:
```diff
@@ -1,7 +1,9 @@
+        <page name='Playground Book with 3rd-party libraries'/>
         <page name='How to export my snippets?'/>
-        <page name='How to use Carbon API?'/>
+        <page name='How to use carbon api?'/>
```
Carbon should always be capitalized. API should always be uppercased, as it is an acronym.

miguelangel-dev replied:
....mmm...but, your suggestions are the opposite, rigth?

truizlop replied:
No, you have changed it and I am asking to revert it to how it was before.

miguelangel-dev replied:
Ahhhhhh!!! I saw it as a “commit suggestion” - it should be happening during merge! Sure, thanks :)

miguelangel-dev replied:
Ok, I see what happens, it is the "place" where playground links its pages, and it continues in lowercased, but as it is not case-sensitive it is fine. Anyway, I have updated it.

truizlop reviewed (changes requested):
Images `airdrop.png` and `tutorial-playgroundbook.jpeg` contain text in Spanish. They should be changed to English.

miguelangel-dev committed:
Update Documentation.app/Contents/MacOS/First steps.playground/Pages/How to export my snippets?.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update Documentation.app/Contents/MacOS/First steps.playground/Pages/How to use Carbon API?.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update Documentation.app/Contents/MacOS/First steps.playground/Pages/Playground Book with 3rd-party libraries.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update Documentation.app/Contents/MacOS/First steps.playground/Pages/Playground Book with 3rd-party libraries.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update Documentation.app/Contents/MacOS/First steps.playground/Pages/Playground Book with 3rd-party libraries.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update Documentation.app/Contents/MacOS/First steps.playground/Pages/Playground Book with 3rd-party libraries.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update Documentation.app/Contents/MacOS/First steps.playground/Pages/Playground Book with 3rd-party libraries.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update Documentation.app/Contents/MacOS/First steps.playground/Pages/Playground Book with 3rd-party libraries.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update Documentation.app/Contents/MacOS/First steps.playground/Pages/Playground Book with 3rd-party libraries.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update Documentation.app/Contents/MacOS/First steps.playground/Pages/Playground Book with 3rd-party libraries.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

comments in PR

update assets to eng

Update Documentation.app/Contents/MacOS/First steps.playground/Pages/How to export my snippets?.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update Documentation.app/Contents/MacOS/First steps.playground/Pages/Playground Book with 3rd-party libraries.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update Documentation.app/Contents/MacOS/First steps.playground/Pages/Playground Book with 3rd-party libraries.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

miguelangel-dev commented:
> Images `airdrop.png` and `tutorial-playgroundbook.jpeg` contain text in Spanish. They should be changed to English.

Applied the changes, thanks ^^

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/Pages/How to export my snippets?.xcplaygroundpage/Contents.swift:
```diff
@@ -55,10 +54,7 @@ In this tutorial, we assume you do not have any project, and we will follow up f
-/*:
- &nbsp;
- ### 4. Customize your snippets!
+ #### Step 4: Customize your snippets!
```
```suggestion
 #### Step 4: Customize your snippets
```

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/Pages/Playground Book with 3rd-party libraries.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,79 @@
+ 
+ #### Step 2: Open `Package.swift` using Xcode
+ 
+ Go to `~/Desktop` and open the `Package.swift` you created in previous steps.
```
```suggestion
 Go to `~/Desktop` and open the `Package.swift` you created in a previous step.
```

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/Pages/Playground Book with 3rd-party libraries.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,79 @@
+ 
+ After the Playground has been created, you can continue from the second step in the previous section in order to load it into the iPad.
+ 
+ 📣 If you download [Swift Playgrounds](https://apps.apple.com/es/app/swift-playgrounds/id1496833156?mt=12) app in your macOS, you can open it and start to work. In the future, you can transfer the Playground Book to your iPad to continue the work; even better, you can use `iCloud Drive` and share the Playground between your Mac and your iPad.
```
```suggestion
 📣 If you download [Swift Playgrounds](https://apps.apple.com/es/app/swift-playgrounds/id1496833156?mt=12) app in your macOS, you can open it and start working. Later, you can transfer the Playground Book to your iPad to continue working; even better, you can use `iCloud Drive` and share the Playground between your Mac and your iPad, keeping its state in sync.
```

truizlop commented on Documentation.app/Contents/MacOS/First steps.playground/Pages/Xcode Playground → nef Playground.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,73 @@
+ 
+ - 🖥 Enables __compilation__ of Playgrounds using command-line.
+ 
+ + 🗞 Renders into __markdown__ files, a __Jekyll__ site or __Carbon__ snippets.
```
```suggestion
 + 🗞 Renders into __Markdown__ files, a __Jekyll__ site or __Carbon__ snippets.
```
